src/ca/noxid/lab/mapdata/Mapdata.java

```diff
@@ -190,8 +198,11 @@ public ByteBuffer toBuf(MOD_TYPE format, String charEncoding) throws Unsupported
 			retVal.put(npcSet1.getBytes(charEncoding), 0, npcSet1.length() < 0x20 ? npcSet1.length() : 0x1F);
 			retVal.position(0x84);
 			retVal.put(npcSet2.getBytes(charEncoding), 0, npcSet2.length() < 0x20 ? npcSet2.length() : 0x1F);
-			retVal.position(0xC4);
+			retVal.position(0xA4);
 			retVal.put((byte) bossNum);
+			retVal.position(0xA5);
+			retVal.put(jpName, 0, 0x1f);
```

Comment:
Can you add the ternary used elsewhere in the event the length of jpName changes?